### ISSUE TITLE
Error in github_resolve_ref.github_release trying to install IRkernel

### ISSUE THREAD

**Issue by breiter** [r-lib/remotes#767]
Installing IRKernel via remotes in [rocker-vesrioned2/scripts/install_jupiter.sh](https://github.com/rocker-org/rocker-versioned2/blob/R4.3.1/scripts/install_jupyter.sh) fails with an error  Cannot find repo repo [IRkernel/IRkernel](https://github.com/IRkernel/IRkernel/) although it exists and cannot open URL '[https://api.github.com/repos/IRkernel/IRkernel/releases](https://api.github.com/repos/IRkernel/IRkernel/releases)' which also exists.

https://github.com/rocker-org/rocker-versioned2/blob/ef593dcd7b334e02e79188a9e17dcf6149c178b9/scripts/install_jupyter.sh#L36

```
R --quiet -e 'remotes::install_github("IRkernel/IRkernel@*release")'
```

```
41.01 > remotes::install_github("IRkernel/IRkernel@*release")
41.22 Error in github_resolve_ref.github_release(meta$ref %||% ref, meta, host = host,  : 
41.22   Cannot find repo IRkernel/IRkernel.
41.22 cannot open URL 'https://api.github.com/repos/IRkernel/IRkernel/releases'
41.22 Calls: <Anonymous> ... github_resolve_ref -> github_resolve_ref.github_release
41.22 Execution halted
------
Dockerfile:97
--------------------
  95 |     
  96 |     # install jupyter
  97 | >>> RUN /rocker_scripts/install_jupyter.sh
  98 |     
  99 |     # set up renv in order to build cache in ~/.cache/R/renv/library by creating a
```

I think this is happening in [install-github.R](https://github.com/r-lib/remotes/blob/88751711e328e24ef9284109b545b5e2668f4bb1/install-github.R#L3962-L3982) in `github_resolve_ref.github_release` but I'm stumped as to why.

```r
  # Retrieve the ref for the latest release
  #' @export
  github_resolve_ref.github_release <- function(x, params, ..., host, auth_token = github_pat()) {
    # GET /repos/:user/:repo/releases
    path <- paste("repos", params$username, params$repo, "releases", sep = "/")
    response <- tryCatch(
      github_GET(path, host = host, pat = auth_token),
      error = function(e) e
    )
  
    if (methods::is(response, "error") || !is.null(response$message)) {
      stop("Cannot find repo ", params$username, "/", params$repo, ".", "\n",
        response$message)
    }
  
    if (length(response) == 0L)
      stop("No releases found for repo ", params$username, "/", params$repo, ".")
  
    params$ref <- response[[1L]]$tag_name
    params
  }
```

**Comment by breiter** [comment]
Working directly on the unaltered rocker image, so I need to find the problem somewhere else.

```
$ docker run --rm -it rocker/rstudio:4 /bin/bash
root@908254633e8c:/# bash /rocker_scripts/install_jupyter.sh 
...
* DONE (IRkernel)
> 
> 
> IRkernel::installspec(user = FALSE)
> 
> 
Check the avalable jupyter kernels...

Available kernels:
  ir         /usr/local/share/jupyter/kernels/ir
  python3    /usr/local/share/jupyter/kernels/python3

Install jupyter, done!
```

thanks!

**Comment by breiter** [comment]
I think this was caused by a regression in a security fix for curl in Ubuntu 22.04.

> ==========================================================================
> Ubuntu Security Notice USN-6237-2
> July 19, 2023
> 
> curl regression
> ==========================================================================
> 
> A security issue affects these releases of Ubuntu and its derivatives:
> 
> - Ubuntu 22.04 LTS
> 
> Summary:
> 
> USN-6237-1 introduced a regression in curl.
> 
> Software Description:
> - curl: HTTP, HTTPS, and FTP client and client libraries
> 
> Details:
> 
> USN-6237-1 fixed vulnerabilities in curl. The update caused a certificate
> wildcard handling regression on Ubuntu 22.04 LTS. This update fixes the
> problem.
> 
> We apologize for the inconvenience.
